ArduCopter/mode.cpp

```diff
@@ -472,6 +472,31 @@ void Copter::notify_flight_mode() {
     notify.set_flight_mode_str(flightmode->name4());
 }
 
+// called when the land detector determines we are landed:
+void Mode::land_complete()
+{
+    // trigger disarm-on-land if configured
+    if ((g.throttle_behavior & THR_BEHAVE_DISARM_ON_LAND_DETECT) == 0) {
+        // not configured to disarm
+        return;
+    }
+
+    if (!motors->armed()) {
+        // not armed
+        return;
+    }
+
+    if (has_manual_throttle()) {
+        return;
+    }
+    // we should probably have a "allows_disarm(...)" callback
+    if (!allows_arming(AP_Arming::Method::LANDING)) {
+        return;
+    }
+
+    AP::arming().disarm(AP_Arming::Method::LANDED);
```

Comment:
How did we end up with both `LANDED` and `LANDING`?  Pre-existing, but we should clarify.

Perhaps `s/LANDING/LAND_DETECTOR` if it makes sense.